client/data/site-settings/use-site-settings-query.ts

```diff
@@ -0,0 +1,15 @@
+import { useQuery } from '@tanstack/react-query';
+import wp from 'calypso/lib/wp';
+
+const useSiteSettingsQuery = ( siteId: number | null ) =>
```

Comment:
In general, it's nice to see this refactored to React Query 👍 

I'm a bit concerned about the duplication though, as all settings pages could take a long time to be refactored.